src/session/session_api.c

```diff
@@ -833,16 +833,42 @@ __session_alter_readonly(WT_SESSION *wt_session, const char *uri, const char *co
 int
 __wt_session_count(WT_SESSION *wt_session, const char *uri, int64_t *count)
 {
+    WT_CKPT ckpt;
     WT_DECL_RET;
+    WT_SESSION_IMPL *session;
+    WT_TABLE *table;
+    int64_t row_count;
+    u_int i;
 
-    WT_UNUSED(uri);
-    WT_UNUSED(wt_session);
+    session = (WT_SESSION_IMPL *)wt_session;
+    table = NULL;
 
-    *count = -1;
+    SESSION_API_CALL_NOCONF(session, count);
+    row_count = *count = -1;
+    if (WT_PREFIX_MATCH(uri, "file:")) {
+        if ((ret = __wt_meta_checkpoint(session, uri, NULL, &ckpt)) != 0)
+            WT_ERR(ret == WT_NOTFOUND ? ENOENT : ret);
+
+        *count = ckpt.ps.row_count;
+    } else if (WT_PREFIX_SKIP(uri, "table:")) {
+        if ((ret = __wt_schema_get_table(session, uri, strlen(uri), false, 0, &table)) != 0)
+            WT_ERR(ret == WT_NOTFOUND ? ENOENT : ret);
+
+        /* Set the number of rows from one column group, sum the column group byte counts. */
```

Comment:
If we only support row count, why we aggregate the bytes counts? I think this is not implemented.

Reply:
Yes, currently the byte count part is not yet implemented.